typegraph/python/typegraph/graph/params.py

```diff
@@ -87,153 +92,113 @@ def oauth2(
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_github(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_github(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_gitlab(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_gitlab(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_google(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_google(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_instagram(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_instagram(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_linkedin(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_linkedin(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_microsoft(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_microsoft(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_reddit(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_reddit(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_slack(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_slack(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_stackexchange(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_stackexchange(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
+
+    def oauth2_twitter(scopes: str) -> "RawAuth":
+        res = wit_utils.oauth2_twitter(store, scopes)
+        if isinstance(res, Err):
+            raise Exception(res.value)
+        return RawAuth(res.value)
 
-oauth2 = dict()
+
+@dataclass
+class RawAuth:
+    json_str: str
 
 
 @dataclass
 class OAuth2Params:
     authorize_url: str
     access_url: str
     profile_url: Optional[str]
-    profiler: Optional[str]
-
-
-_oauth2_params = {
-    "digitalocean": OAuth2Params(
-        authorize_url="https://cloud.digitalocean.com/v1/oauth/authorize",
-        access_url="https://cloud.digitalocean.com/v1/oauth/token",
-        # https://docs.digitalocean.com/reference/api/api-reference/#operation/account_get
-        profile_url="https://api.digitalocean.com/v2/account",
-        profiler="(p) => ({id: p.account.uuid})",
-    ),
-    "discord": OAuth2Params(
-        authorize_url="https://discord.com/api/oauth2/authorize",
```

Comment:
I think it might be because the diff is too large?
https://github.com/metatypedev/metatype/blob/MET-238-sdk-auth/typegraph/core/src/utils/mod.rs#L209

Reply:
@afmika indeed, let's make it more general or macro based (along the the wit bindings change) 👍